Including instrumented_mutex.h leads to compilation errors

When adding an include of instrumented_mutex.h in write_buffer_manager.h compilation fails:
```
#include "rocksdb/cache.h"
#include "monitoring/instrumented_mutex.h"

namespace ROCKSDB_NAMESPACE {
class CacheReservationManager;
```

When adding the same to the write_buffer_manager.cpp compilation passes. This will probably happen with other header files, I just stumbled on this while modifying write_buffer_manager.h.
This implies some issue somewhere in the include chain:
```
udi@udi-speedb:~/speedb-oss-OTHER$ make memtable/write_buffer_manager.o
$DEBUG_LEVEL is 1
Makefile:165: Warning: Compiling in debug mode. Don't use the resulting binary in production
* GEN     make_config.mk
  CC       memtable/write_buffer_manager.o
In file included from ./port/port_posix.h:16,
                 from ./port/port.h:18,
                 from ./monitoring/statistics.h:16,
                 from ./monitoring/instrumented_mutex.h:8,
                 from ./include/rocksdb/write_buffer_manager.h:22,
                 from memtable/write_buffer_manager.cc:10:
./include/rocksdb/options.h:886:19: error: ‘WriteBufferManager’ was not declared in this scope
  886 |   std::shared_ptr<WriteBufferManager> write_buffer_manager = nullptr;
      |                   ^~~~~~~~~~~~~~~~~~
./include/rocksdb/options.h:886:37: error: template argument 1 is invalid
  886 |   std::shared_ptr<WriteBufferManager> write_buffer_manager = nullptr;
      |                                     ^
./include/rocksdb/options.h:886:62: error: cannot convert ‘std::nullptr_t’ to ‘int’ in initialization
  886 |   std::shared_ptr<WriteBufferManager> write_buffer_manager = nullptr;
      |                                                              ^~~~~~~
make: *** [Makefile:2422: memtable/write_buffer_manager.o] Error 1
```

As a workaround/patch, I managed to resolve the compilation issue of write_buffer_manager file by forward declaring WriteBufferManager:
```
#include "rocksdb/cache.h"
namespace ROCKSDB_NAMESPACE {
class WriteBufferManager;

}
#include "monitoring/instrumented_mutex.h"
```
However, that causes failure of multiple other files including write_buffer_manager.h

--- a/include/rocksdb/options.h
+++ b/include/rocksdb/options.h
@@ -29,7 +29,6 @@
 #include "rocksdb/types.h"
 #include "rocksdb/universal_compaction.h"
 #include "rocksdb/version.h"
-#include "rocksdb/write_buffer_manager.h"
 
 #ifdef max
 #undef max
@@ -55,6 +54,7 @@ class Slice;
 class Statistics;
 class InternalKeyComparator;
 class WalFilter;
+class WriteBufferManager;
 class FileSystem;
 class WriteController;
 

--- a/options/db_options.cc
+++ b/options/db_options.cc
@@ -22,6 +22,7 @@
 #include "rocksdb/system_clock.h"
 #include "rocksdb/utilities/options_type.h"
 #include "rocksdb/wal_filter.h"
+#include "rocksdb/write_buffer_manager.h"
 #include "util/string_util.h"
 
 namespace ROCKSDB_NAMESPACE {

--- a/options/options.cc
+++ b/options/options.cc
@@ -30,6 +30,7 @@
 #include "rocksdb/table.h"
 #include "rocksdb/table_properties.h"
 #include "rocksdb/wal_filter.h"
+#include "rocksdb/write_buffer_manager.h"
 #include "table/block_based/block_based_table_factory.h"
 #include "util/compression.h"
 

